tests/integration/targets/keyring/tasks/main.yml

```diff
@@ -0,0 +1,70 @@
+---
+- name: Ensure required packages for headless keyring access are installed (RPM)
+  ansible.builtin.package:
+      name: gnome-keyring
+  become: true
+  when: "'localhost' not in inventory_hostname"
+
+- name: Ensure keyring is installed (RPM)
+  ansible.builtin.dnf:
+    name: python3-keyring
+    state: present
+  become: true
+  when: ansible_facts['os_family'] == 'RedHat'
+
+- name: Ensure keyring is installed (pip)
+  ansible.builtin.pip:
+    name: keyring
+    state: present
+  become: true
+  when: ansible_facts['os_family'] != 'RedHat'
+
+# Set password for new account
+# Expected result: success
+- name: Set password for test/test1
+  community.general.keyring:
+    service: test
+    username: test1
+    user_password: "{{ user_password }}"
+    keyring_password: "{{ keyring_password }}"
+  register: set_password
+
+# Print out password to confirm it has been set
+# Expected result: success
+- name: Retrieve password for test/test1
+  community.general.keyring_info:
+    service: test
+    username: test1
+    keyring_password: "{{ keyring_password }}"
+  register: test_set_password
```

Comment:
I don't see any test (`assert` statements) that check whether this module actually does what was expected. This module could return complete garbage and the tests won't notice.

Reply:
Yeah, I hadn't quiet worked out how I wanted to do the tests.
I was thinking about running them as part of the CI, but I wasn't sure how the Python deps would be handled.

I'll add `assert` statements to test the functionality

Reply:
In CI you need to install the Python dependencies you need.